### PR TITLE
Make tests `pml_x_psatd` and `pml_x_yee` also restart tests.

### PR DESCRIPTION
**Pull request by Yin-YinjianZhao**
Description edited by @RemiLehe :

This PR makes the `pml_x_psatd` and `pml_x_yee` restart tests. We chose a restart iteration that corresponds to a timestep at which the pulses reached the PML. 

An important bug in PML is also fixed as part of this PR: The variable `do_pml` was not properly initialized when doing a restart.

### CODE REVIEW

**Comment by Yin-YinjianZhao** [comment]
Based on the current modification in this PR, it is found that:
1) This print does not show up https://github.com/ECP-WarpX/WarpX/pull/2444/files#diff-d13fa6f436037fcf4b92f52a18ff13183f11240d366985e5847711a80aec971aR1203.
2) This print also does not show up https://github.com/ECP-WarpX/WarpX/pull/2444/files#diff-57fd056f54b275309adcce9674a51dd568f0cb17dcd56d4c24188fdf8d3e3b46R330.
This maybe suggests that there is a bug when reading the PML.

Because the PML is not well done in the restart, the fields after the restart are very different from the those without restart.

For example:
Below is Bx at time step 200 after restart.
pml_x_psatd_plt00200
Below is Bx at time step 200 without restart.
pml_x_psatd_plt00200 old 7500000

**Comment by EZoni** [comment]
@Yin-YinjianZhao Thank you for the PR, Yin! For debugging purposes, I think it could be also useful to copy/adapt the analysis done in this script Examples/Tests/restart/analysis_restart.py into `analysis_pml_psatd.py`, so that we can maybe see if certain grid fields and/or particle attributes happen to be more mismatched than others.

**Comment by Yin-YinjianZhao** [comment]
> @Yin-YinjianZhao Thank you for the PR, Yin! For debugging purposes, I think it could be also useful to copy/adapt the analysis done in this script Examples/Tests/restart/analysis_restart.py into `analysis_pml_psatd.py`, so that we can maybe see if certain grid fields and/or particle attributes happen to be more mismatched than others.

@EZoni Thanks for the suggestion! But Remi just found the bug. We may have solved the problem.

File: Examples/Tests/PML/analysis_pml_psatd.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
```
Comment (RemiLehe) [inline Examples/Tests/PML/analysis_pml_psatd.py]:
```suggestion
#! /usr/bin/env python
```

File: Examples/Tests/PML/analysis_pml_yee.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
```
Comment (RemiLehe) [inline Examples/Tests/PML/analysis_pml_yee.py]:
```suggestion
#! /usr/bin/env python
```

**Comment by Yin-YinjianZhao** [comment]
Bug fixed, and everything seems working fine now.

**Comment by EZoni** [comment]
Thank you for the PR, Yin! I wonder if we should make the part of Python analysis that checks the restart output become a separate Python module that is available in the repo and can be imported. This could be done in a separate PR, too. I'm just pointing this out because with this PR that code will be repeated three times (twice for the tests modified here and once for the restart tests that use a different analysis script).

**Comment by EZoni** [comment]
Also, I will not close issue #2399 because I tried this PR but still saw some relatively large errors as described in the issue. They probably have something to do with the fact that the Galilean algorithm is not yet supported with PMLs, so they are unrelated to the issue fixed here.

**Review by RemiLehe — APPROVED** [review]

**Comment by RemiLehe** [comment]
I think I will merge for now, because of the imporant bug fix that this PR introduces. 
The Python clean-up could be probably done in a follow-up PR.

**Pull request merged**